### PR TITLE
Add Helm chart for AKS managed cluster

### PR DESCRIPTION
**Pull request by Jont828**
Add Helm chart for AKS managed cluster

Note: this is originated from the following repo: https://github.com/nellyk/azure-managed-cluster-capz-helm

Co-authored-by: Nelly Kiboi <3062772+nellyk@users.noreply.github.com>
Co-authored-by: Alessandro Vozza <alessandro.vozza@microsoft.com>

### CODE REVIEW

File: charts/azure-managed-cluster/Chart.yaml
Comment (mboersma) [inline charts/azure-managed-cluster/Chart.yaml:1]: We'll need to add a `maintainers:` field here to keep the linter happy, and its `name` should be just a GitHub handle.

See https://github.com/mboersma/cluster-api-charts/pull/3#issuecomment-2098980928

**Comment by mboersma** [comment]
Some whitespace cleanup required before the linter is happy:

```
charts/azure-managed-cluster/values.yaml
  Error: 10:1 [trailing-spaces] trailing spaces
  Error: 21:16 [trailing-spaces] trailing spaces
  Error: 45:1 [trailing-spaces] trailing spaces
  Error: 57:17 [colons] too many spaces after colon
  Error: 61:16 [colons] too many spaces before colon
  Error: 61:18 [trailing-spaces] trailing spaces
  Error: 77:26 [trailing-spaces] trailing spaces
  Error: 79:10 [trailing-spaces] trailing spaces
  Error: 83:21 [trailing-spaces] trailing spaces
  Error: 106:6 [indentation] wrong indentation: expected 4 but found 5
  Error: 108:12 [colons] too many spaces before colon
  Error: 111:2 [comments] missing starting space in comment
  Error: 111:43 [trailing-spaces] trailing spaces
  Error: 115:10 [trailing-spaces] trailing spaces
  Error: 117:16 [comments] too few spaces before comment
  Error: 125:23 [trailing-spaces] trailing spaces
  Error: 126:6 [indentation] wrong indentation: expected 6 but found 5
  Error: 1[30](https://github.com/mboersma/cluster-api-charts/actions/runs/9025955495/job/24818320639?pr=4#step:7:31):6 [indentation] wrong indentation: expected 6 but found 5
  Error: 1[31](https://github.com/mboersma/cluster-api-charts/actions/runs/9025955495/job/24818320639?pr=4#step:7:32):19 [trailing-spaces] trailing spaces
  Error: 147:22 [trailing-spaces] trailing spaces
```

File: charts/azure-managed-cluster/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v2
+name: azure-managed-cluster
+description: A Helm chart to deploy an AKS cluster via the cluster-api-provider-azure ManagedCluster
+type: application
+maintainers:
+- email: jt572@cornell.edu
+  name: Jonathan Tong
```
Comment (mboersma) [inline charts/azure-managed-cluster/Chart.yaml]:
```suggestion
  name: Jont828
```
As I mentioned [elsewhere](https://github.com/mboersma/cluster-api-charts/pull/3#issuecomment-2098980928), `helm-lint` is strict here and wants a GitHub handle.

Comment (Jont828) [reply charts/azure-managed-cluster/Chart.yaml]: It seems odd to me that the `name` field isn't actually the name. I think it makes more sense than the GitHub handle as you already have the email. Can we change the linter settings?

Comment (mboersma) [reply charts/azure-managed-cluster/Chart.yaml]: We could shut off the `maintainers` check but we decided not to in the conversation I linked to above.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+*.code-workspace
```
Comment (dtzar) [inline .gitignore:1]: Do we want to keep this or perhaps provide a more comprehensive git ignore file on a separate PR?

**Review by dtzar — APPROVED** [review]
Can we also please get a very basic readme in the root of this chart?

File: charts/azure-managed-cluster/values.yaml [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+  virtualNetwork: {}
+  disableLocalAccounts: false
+  # Only 1 Nodepool of mode System is allowed
+
+agentPools: []
+
+defaults:
+  agentPool:
+    mode: User
+    # TODO: should mode be set or required?
```
Comment (mboersma) [inline charts/azure-managed-cluster/values.yaml:118]: Is this question still open?

File: charts/azure-managed-cluster/values.yaml [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+  clusterIdentitySecretValueB64: ''
+  resourceID: ""
+  type: ServicePrincipal
+
+controlplane:
+
+  ## SSH public key (must be valid)
+
+  ## Kubernetes version
+  kubernetes_version: "v1.28.5"
```
Comment (mboersma) [inline charts/azure-managed-cluster/values.yaml:47]:
```suggestion
  kubernetes_version: "v1.28.9"
```

**Review by mboersma — COMMENTED** [review]
lgtm overall

**Comment by jackfrancis** [comment]
lgtm

let's merge this and address any follow-up items in a follow-up PR

rapid iteration ftw

**Comment by mboersma** [comment]
```shell
helm repo add capi https://mboersma.github.io/cluster-api-charts
helm install <name> capi/azure-managed-cluster
```

**Pull request merged**